Use correct signatures for METH_NOARGS.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45528 Use correct signatures for METH_NOARGS.**

As described in https://github.com/pytorch/pytorch/issues/45419,
resolving a bunch of cpython signature issues.

#Closes: https://github.com/pytorch/pytorch/issues/45419

Differential Revision: [D24000626](https://our.internmc.facebook.com/intern/diff/D24000626/)

pritamdamania committed:
Use correct signatures for METH_NOARGS.

As described in https://github.com/pytorch/pytorch/issues/45419,
resolving a bunch of cpython signature issues.

#Closes: https://github.com/pytorch/pytorch/issues/45419

Differential Revision: [D24000626](https://our.internmc.facebook.com/intern/diff/D24000626/)

[ghstack-poisoned]

albanD commented:
Thanks for the PR! This is indeed important to fix!

pritamdamania committed:
Update on "Use correct signatures for METH_NOARGS."


As described in https://github.com/pytorch/pytorch/issues/45419,
resolving a bunch of cpython signature issues.

#Closes: https://github.com/pytorch/pytorch/issues/45419

Differential Revision: [D24000626](https://our.internmc.facebook.com/intern/diff/D24000626/)

[ghstack-poisoned]

albanD commented on torch/csrc/Module.cpp:
```diff
@@ -608,10 +608,10 @@ static PyMethodDef TorchMethods[] = {
   {"set_flush_denormal", (PyCFunction)THPModule_setFlushDenormal, METH_O,     nullptr},
   {"get_default_dtype", (PyCFunction)THPModule_getDefaultDtype, METH_NOARGS,  nullptr},
   {"_get_default_device", (PyCFunction)THPModule_getDefaultDevice, METH_NOARGS,   nullptr},
```
nit:  I guess the cast could be removed here as well?
(same for the other ones below)

albanD reviewed (approved):
lgtm thanks for the update.

pritamdamania committed:
Update on "Use correct signatures for METH_NOARGS."


As described in https://github.com/pytorch/pytorch/issues/45419,
resolving a bunch of cpython signature issues.

#Closes: https://github.com/pytorch/pytorch/issues/45419

Differential Revision: [D24000626](https://our.internmc.facebook.com/intern/diff/D24000626/)

[ghstack-poisoned]
